cli/js/net.ts

```diff
@@ -109,63 +110,56 @@ export class ConnImpl implements Conn {
 }
 
 export class ListenerImpl implements Listener {
-  constructor(
-    readonly rid: number,
-    readonly addr: Addr,
-    private closing: boolean = false
-  ) {}
+  private closed = false;
+
+  constructor(readonly rid: number, readonly addr: Addr) {}
 
   async accept(): Promise<Conn> {
+    if (this.closed) {
+      // This should match the error thrown by op_accept if closing.
+      throw new errors.Other("Listener has been closed");
+    }
     const res = await sendAsync("op_accept", { rid: this.rid });
     return new ConnImpl(res.rid, res.remoteAddr, res.localAddr);
   }
 
   close(): void {
-    this.closing = true;
+    this.closed = true;
     close(this.rid);
   }
 
-  async next(): Promise<IteratorResult<Conn>> {
-    if (this.closing) {
-      return { value: undefined, done: true };
-    }
-    return await this.accept()
-      .then(value => ({ value, done: false }))
-      .catch(e => {
-        // It wouldn't be correct to simply check this.closing here.
-        // TODO: Get a proper error kind for this case, don't check the message.
-        // The current error kind is Other.
-        if (e.message == "Listener has been closed") {
-          return { value: undefined, done: true };
+  async *[Symbol.asyncIterator](): AsyncIterator<Conn> {
+    while (true) {
+      try {
+        yield await this.accept();
+      } catch (error) {
+        // This should match the error thrown by op_accept if closing.
+        if (error.message == "Listener has been closed") {
+          break;
         }
-        throw e;
-      });
-  }
-
-  [Symbol.asyncIterator](): AsyncIterator<Conn> {
-    return this;
+        throw error;
+      }
+    }
   }
 }
 
-export async function recvfrom(
-  rid: number,
-  p: Uint8Array
-): Promise<[number, Addr]> {
-  const { size, remoteAddr } = await sendAsync("op_receive", { rid }, p);
-  return [size, remoteAddr];
-}
```

Comment:
I guess it makes sense not to expose this as top level function

Reply:
CC @ry please vet

Reply:
Restored. But I'm not sure, it's not tested in isolation and it has a weird name.

Reply:
Actually I see https://linux.die.net/man/2/recvfrom. Nevermind.